src/non_RL_algos/shortestPath.py

```diff
@@ -0,0 +1,250 @@
+import argparse
+import logging
+import os
+import random
+from collections import defaultdict
+from datetime import datetime
+from pathlib import Path
+from shutil import copyfile
+
+from common.common_functionalities import normalize_scheduling_probabilities, create_input_file
+from siminterface.simulator import Simulator
+from spinterface import SimulatorAction
+
+log = logging.getLogger(__name__)
+DATETIME = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+
+
+def copy_input_files(target_dir, network_path, service_path, sim_config_path):
+    """Create the results directory and copy input files"""
+    new_network_path = f"{target_dir}/{os.path.basename(network_path)}"
+    new_service_path = f"{target_dir}/{os.path.basename(service_path)}"
+    new_sim_config_path = f"{target_dir}/{os.path.basename(sim_config_path)}"
+
+    os.makedirs(target_dir, exist_ok=True)
+    copyfile(network_path, new_network_path)
+    copyfile(service_path, new_service_path)
+    copyfile(sim_config_path, new_sim_config_path)
+
+
+def get_ingress_nodes_and_cap(network):
```

Comment:
How about we make this a utility function too? Similar to `normalize_scheduling_probabilities`. Then we don't have douplicate code in each algorithm

Reply:
Done to many more functions that were common